pkg/version/compare_test.go

```diff
@@ -334,6 +334,13 @@ func TestCompareAnyVersion(t *testing.T) {
 			versionB:       "3.0.61.1",
 			expectedResult: 1,
 		},
+		{
+			// Even though 4.6.1 > 4.6.0, dev builds are always considered greater than release builds
+			// with the same x.y (i.e. major and minor) numbers.
+			versionA:       "4.6.1",
+			versionB:       "4.6.0-3-gabc1233456",
```

Comment:
Thinking about this again, I don't think this test case is valid _after we update the release workflow_.  Assuming we start using the new dev versioning scheme when we create the 4.5 release branch, then this test case will be valid for versions < 4.6 (there would be no 4.5 dev versions).

Reply:
Silly question that I probably should know.  What is the new dev versioning scheme?

In this case would the dev version become `4.6.1-3-gabc1233456` when `4.6.1` is released?

Reply:
> What is the new dev versioning scheme?

Explained in this document: https://docs.google.com/document/d/1zPt_yDy84lXPhf-c84sRAoTJKfRLTbxv8-6Cu7e8S5w/edit#heading=h.xy5bla31nvma. TLDR: like today, replace `major.minor.x-...` -> `major.minor.0-...` and when we create the release branch `major.minor`, we will bump master to start producing `major.(minor+1)` tags.

> In this case would the dev version become 4.6.1-3-gabc1233456 when 4.6.1 is released?

Depends where you look. 
On the master branch, commits before the 4.6 release branch was created will be 4.6.0-3-gabc1233456. After the branch was created, they will look like `4.7.0-3-gabc123456`.
On the release branch, after 4.6.1 is released, the commits will look like `4.6.1-3-gabc1233456`, before they will look like 4.6.0-1-gabc1233456.

Reply:
A small note that commits on master will always have "patch" equal to `0`: 4.6.**0**-(n)-g(sha), 4.7.**0**-(n)-g(sha), 4.8.**0**-(n)-g(sha) and so on. This makes them _less_ according to SemVer rules than corresponding release numbers:
4.6.0-(n)-g(sha) < 4.6.0 < 4.6.1 < ...
4.7.0-(n)-g(sha) < 4.7.0 < 4.7.1 < ...